### PR TITLE
Fix aggregates

### PR DESCRIPTION
**Pull request by bdtomlin**
Addresses issues with aggregate queries

See #292, #304, and #287

### CODE REVIEW

File: spec/query_spec.cr [hunk 1]
```diff
@@ -474,27 +477,15 @@ describe Avram::Query do
...
-    end
-
-    it "works with chained where clauses" do
-      scores = [100.4, 123.22]
-      sum = scores.sum
-      out_of_range = 200
-      scores.each { |score| UserBox.create &.average_score(score) }
-      query_sum = UserQuery.new.average_score.lt(out_of_range).average_score.select_sum
-      query_sum.should eq sum
-    end
```
Comment (paulcsmith) [inline spec/query_spec.cr]: I think this test would still be nice to have but maybe there is a reason for deleting it? Is it covered by another test?

File: spec/query_spec.cr [hunk 2]
```diff
@@ -398,72 +414,59 @@ describe Avram::Query do
...
     end
   end
 
-  describe "#select_sum! for Int64 column" do
+  describe "#select_sum! for Int32 column" do
     it "returns 0 if there are no records" do
       query_sum = UserQuery.new.id.select_sum!
-      query_sum.should eq 0_i64
+      query_sum.should eq 0
+      query_sum.should be_a Int64
```
Comment (paulcsmith) [inline spec/query_spec.cr]: Nice catch. More robust to check the type 👍

File: spec/query_spec.cr (hunk 1 above)
Comment (bdtomlin) [reply spec/query_spec.cr]: Yes, it's covered in a different test. I created a generic `describe "#select_sum" do` that handles the 2 non type specific tests here: https://github.com/luckyframework/avram/blob/69259484eac9995277a3384a6ac99d981ac09d26/spec/query_spec.cr#L402-L415

It seemed unnecessary to repeat for every type of integer and float. Let me know if you would like me to do something a little different.

Comment (paulcsmith) [reply spec/query_spec.cr]: Ah I see! I think this was a good call 👍

**Comment by paulcsmith** [comment]
Looks awesome. Loving these new fixes and improved/more accurate type signatures in aggregates

**Pull request merged**